### PR TITLE
Setup bancho-service to run containerized

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

**Review by tsunyoku — COMMENTED** [review]
unsure if i agree with making configs not use DSN, it removes some freedom imo for no gain. otherwise fine tho

**Comment by cmyui** [comment]
@tsunyoku not sure how it removes any freedom - you just more easily have access to the underlying subparts, while still constructing the dsn & storing it under the settings anyways :o

File: settings.py [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
 cfg = Config(".env")
 
-SERVER_DOMAIN: str = cfg("SERVER_DOMAIN")
-SERVER_PORT: int = cfg("SERVER_PORT", cast=int)
+READ_DB_USER = cfg('READ_DB_USER')
+READ_DB_PASS = cfg('READ_DB_PASS')
+READ_DB_HOST = cfg('READ_DB_HOST')
+READ_DB_PORT = cfg('READ_DB_PORT', cast=int)
+READ_DB_NAME = cfg('READ_DB_NAME')
+READ_DB_DSN = 'postgresql://{}:{}@{}:{}/{}?sslmode=prefer'.format(
```
Comment (cmyui) [inline settings.py]: oops postgresql lol

**Comment by tsunyoku** [comment]
> @tsunyoku not sure how it removes any freedom - you just more easily have access to the underlying subparts, while still constructing the dsn & storing it under the settings anyways :o

DSNs are intentionally very configurable i don’t see why we should split our config out from them :o

File: settings.py (hunk 1 above)
Comment (tsunyoku) [reply settings.py]: real

**Comment by cmyui** [comment]
we'll need to setup a healthcheck to ensure the services are prepared before starting the web apps themselves and generally that's just done with a tcp check - that's the usecase i had in mind - for that, we just want `${HOST}:${PORT}`

**Comment by cmyui** [comment]
https://github.com/vishnubob/wait-for-it

**Comment by tsunyoku** [comment]
> we'll need to setup a healthcheck to ensure the services are prepared before starting the web apps themselves and generally that's just done with a tcp check - that's the usecase i had in mind - for that, we just want `${HOST}:${PORT}`

i think that’s a separate issue tho, we shouldn’t have to effect our config like that. if we’re dockerising the service we can have services depend on eachother anyway :p

**Comment by cmyui** [comment]
> if we’re dockerising the service we can have services depend on eachother anyway :p

not quite - docker compose will only check for service startup, it provides no guarantee that the services (e.g. a database) have actually started their tcp listeners (i've already added the dependent services in docker-compose)

> i think that’s a separate issue tho, we shouldn’t have to effect our config like that

these are environmental variables - not necessarily just service configuration for this bancho-service, but rather anything running on the container

**Comment by tsunyoku** [comment]
> these are environmental variables - not necessarily just service configuration for this bancho-service, but rather anything running on the container

i know, i just don’t really like the idea of removing a configurable DSN env var just to check for a service being ready.. i think we should find another way around it

**Comment by cmyui** [comment]
don't get me wrong either - i agree that a dsn is a nice simplification for configuration of a web service, but to me they're an application layer concept while the environmental variables are not - that's why i construct the dsn(s) from the subcomponents when parsing from env vars into our app

if you're able to find a nicer solution i'm open to it, this is just what i've learned form my seniors so far

**Comment by tsunyoku** [comment]
> but to me they're an application layer concept while the environmental variables are not

i would agree if not for DSNs containing config options outside of just the port etc… removing the DSN as the env var removes being able to configure those without hardcoding, kinda sucks

**Comment by tsunyoku** [comment]
that is already apparent in this pr for example, you hardcode the `sslmode` config option

**Comment by cmyui** [comment]
> removing the DSN as the env var removes being able to configure those without hardcoding, kinda sucks

hmm that is actually true - only once i've used so far is `preferssl` in postgres

**Comment by cmyui** [comment]
my plan was to add a bash startup script which would first use [wait-for-it](https://github.com/vishnubob/wait-for-it) for mysql, redis, and rabbitmq before starting the python process to remove the possibility of a race condition (strategy we use at my workplace)

if you'd like, feel free to try finding a nicer solution - i'll continue to bootstrap the infra & score-service for the time being as this isn't really worth spending too much time on rn imo (we can come back to this later too)

**Comment by cmyui** [comment]
note that we also do not need to parse the individual params into settings at the application layer - i'm fine constructing directly from env vars to clean up a bit - i only need those env vars at the container level with my current plan

**Pull request merged**